### PR TITLE
Remove Makefile

### PR DESCRIPTION
**Pull request by jkeiser**
OK, I think we're at parity. The only targets in make that are missing from cmake are:

* `make benchmark`: runs scripts/parse.sh and scripts/parseandstat.sh
* `make benchfeatures` and related scripts: these are presently not compiling, I need to get them back to normal

This removes the Makefile and updates CI not to use it. It will also update HACKING.md when I get to that (thus WIP).

### CODE REVIEW

**Comment by lemire** [comment]
That's fine.

**Comment by jkeiser** [comment]
Why in the world would `apt-get install clang cmake` cause gcc to be installed??

**Comment by jkeiser** [comment]
This also reduces the number of Drone jobs (without, hopefully, reducing coverage).

Minus arm64, everything seems to be working here.

File: .drone.yml [hunk 1]
```diff
@@ -1,559 +1,325 @@
...
-    - mkdir build && cd build
+    - apt-get update -qq
+    - apt-get install -y clang cmake git
+    - mkdir build
+    - cd build
     - cmake $CMAKE_FLAGS ..
-    - make -j
-    - ASAN_OPTIONS="detect_leaks=0" ctest -j4 --output-on-failure -E checkperf
+    - cmake --build . $BUILD_FLAGS
+    - ASAN_OPTIONS="detect_leaks=0" ctest $CTEST_FLAGS -L acceptance -LE per_implementation
```
Comment (lemire) [inline .drone.yml:268]: what is this detect_leaks=0 about?

**Comment by lemire** [comment]
I am not convinced that deleting  examples/quickstart/Makefile is warranted, as this seemed like reasonable documentation... I don't mind if you delete it, but do consider that this can be a useful file to some.

File: .drone.yml [hunk 2]
```diff
@@ -1,559 +1,325 @@
...
+    - ctest $CTEST_FLAGS
 ---
 kind: pipeline
-name: amd64_gcc_cmake_sanitize_noavx
-
-platform:
-  os: linux
-  arch: amd64
-
+name: stylecheck
```
Comment (lemire) [inline .drone.yml:317]: I wonder whether this even worked... I don't remember getting any warning whatsoever regarding style violations... Not that I mind...

File: .drone.yml (hunk 1 above)
Comment (jkeiser) [reply .drone.yml:268]: I really don't know--it was on the command lines for sanitize before too :)

**Comment by jkeiser** [comment]
I think I can do with not deleting examples/quickstart/Makefile.

**Comment by jkeiser** [comment]
THERE. OK. One build system. That was a bit more work than I thought it would be :)

**Comment by lemire** [comment]
F*ck yeah.

File: .drone.yml (hunk 2 above)
Comment (jkeiser) [reply .drone.yml:317]: Now we can start thinking about what to remove and what to dedup between drone and circle!

**Pull request merged**